### PR TITLE
Tooltip in RangeSlider

### LINKED ISSUE #284: In DynamicRange Silder Tooltip is not present in V2.0.0
**Issue Type:**
 enhancement' issue. 

**Platform:**
Web

**Description:**
In DynamicRange Silder Tooltip is not there, So how can user know that which range is selected ??? 
In V1 version tooltip is present. but not in v2.0.0

**Screenshots:**
dynamic-tooltip

**Reactivesearch version:** 
https://github.com/appbaseio/reactivesearch/releases/tag/v2.0.0  ------> I m using this version right now 

**Browser:** [all | Chrome XX | Firefox XX | Edge XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

### PR DESCRIPTION
**Pull request by jyash97**
`Rheostat` has method called `handle` to create custom Handle, which gets following values :
 aria-valuenow , aria-valuemax ,aria-valuemin, className, style and other methods.

Used this handle method which conditionally renders Component if user wants to have tooltip, he can use `handleTooltip` render prop:

```js
<RangeSlider
    handleTooltip = (children, value) => <Tooltip value={value} >{children}</Tooltip>
/>
```
If this render prop is not present we can render the default `Handle` component.
In this way we wont need to add `Tooltip` component in RS and user can use his own Tooltip.

## Issue
I tried doing the same was getting the dom element but it was not visible on UI.

Will close #284

### CODE REVIEW

**Comment by jyash97** [comment]
New API : 
```js 
<RangeSlider 
  showTooltip 
/>
```

Tooltip visible on `active`, `hover` and `focus` state of handle.

tooltip_slider

If this looks good I can go on to add them in other Slider Components

**Comment by davidklebanoff** [comment]
This is looking great! I do feel like from a usability perspective, there may be times when someone wants the tooltips to show even when not hover/focused. Especially for smaller ranges like 1-20. It may be worth adding a configuration, or at least some example code in the docs to allow such behavior.

**Comment by divyanshu013** [comment]
That's a good point. Lets make this prop an `enum`. Lets have a single prop say, `tooltipTrigger`. It can have the following values:

- `focus` (only displays when in focus)
- `hover` (displays when hovered - this would also work in case of focus when mouse is used)
- `always` (always displays the tooltip)
- `none` (default - does not display tooltip. This also helps avoid any breaking changes in the API)

**Comment by metagrover** [comment]
`tooltipTrigger` sounds good to me! 💯

**Comment by jyash97** [comment]
Removed `showTooltip` instead introduced a new prop called `tooltipTrigger` as discussed above.
`tooltipTrigger` have values `none`, `hover`, `always` and `focus`. `none` by default.

---
 **ALWAYS**

tooltip_always

---
**FOCUS**

tooltip_focus

---
**HOVER**
tooltip_hover

---
**NONE**
tooltip_none

File: packages/maps/src/components/basic/GeoDistanceSlider.js [hunk 1]
```diff
@@ -484,6 +495,7 @@ GeoDistanceSlider.propTypes = {
 	react: types.react,
 	showFilter: types.bool,
 	showIcon: types.bool,
+	tooltipTrigger: types.string,
```
Comment (metagrover) [inline packages/maps/src/components/basic/GeoDistanceSlider.js]: We should use `oneOf` prop-type here. You can add it to the reactivecore/types.

File: packages/web/src/components/range/DynamicRangeSlider.js [hunk 2]
```diff
@@ -413,6 +424,7 @@ DynamicRangeSlider.propTypes = {
 	react: types.react,
 	showHistogram: types.bool,
 	showFilter: types.bool,
+	tooltipTrigger: types.string,
```
Comment (metagrover) [inline packages/web/src/components/range/DynamicRangeSlider.js]: update prop-type.

File: packages/web/src/components/range/RangeSlider.js [hunk 3]
```diff
@@ -338,6 +349,7 @@ RangeSlider.propTypes = {
 	histogramQuery: types.func,
 	showFilter: types.bool,
 	showSlider: types.bool,
+	tooltipTrigger: types.string,
```
Comment (metagrover) [inline packages/web/src/components/range/RangeSlider.js]: update prop-type.

File: packages/web/src/styles/Slider.js [hunk 4]
```diff
@@ -52,6 +52,48 @@ const Slider = styled('div')`
 		top: -10px;
 	}
 
+	.slider-tooltip,.slider-tooltip-focus,.slider-tooltip-visible {
```
Comment (metagrover) [inline packages/web/src/styles/Slider.js]: add proper spacing here

File: packages/web/src/styles/Slider.js [hunk 5]
```diff
@@ -52,6 +52,48 @@ const Slider = styled('div')`
...
+    visibility: hidden;
+    background-color: black;
+    color: #fff;
+    text-align: center;
+    border-radius: 6px;
+    padding: 6px 8px;
+    position: absolute;
+    z-index: 1;
+		transform: translate(-50%,-170%); /* Positions Tooltip Container */
```
Comment (metagrover) [inline packages/web/src/styles/Slider.js]: linting issue

File: packages/web/src/styles/Slider.js [hunk 6]
```diff
@@ -52,6 +52,48 @@ const Slider = styled('div')`
...
+	}
+
+	.slider-tooltip-visible {
+		visibility: visible;
+	}
+
+	.slider-tooltip:after,.slider-tooltip-focus:after,.slider-tooltip-visible:after{
+		content: '';
```
Comment (metagrover) [inline packages/web/src/styles/Slider.js]: linting issue + add proper spacing.

**Pull request merged**